pkg/ctl/utils/export.go

```diff
@@ -0,0 +1,273 @@
+		fs.StringVarP(&output, "output", "o", "yaml", "specifies the output format (valid option: json, yaml)")
+		fs.StringVar(&exportDir, "dir", defaultExportDir(), "the directory to save exported files into")
+		fs.BoolVar(&offline, "offline", offline, "disable AWS EC2 AMI lookup and other AWS API interactions")
+		cmdutils.AddVersionFlag(fs, cfg.Metadata, "")
+		cmdutils.AddRegionFlag(fs, p)
+		cmdutils.AddConfigFileFlag(&clusterConfigFile, fs)
+	})
+
+	group.InFlagSet("Initial nodegroup", func(fs *pflag.FlagSet) {
+		fs.StringVar(&ng.Name, "nodegroup-name", "", fmt.Sprintf("name of the nodegroup (generated if unspecified, e.g. %q)", exampleNodeGroupName))
+		fs.BoolVar(&withoutNodeGroup, "without-nodegroup", false, "if set, initial nodegroup will not be created")
+	})
+
+	cmdutils.AddCommonFlagsForAWS(group, p, true)
+
+	group.AddTo(cmd)
+
+	return cmd
+}
+
+func doExport(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg string, cmd *cobra.Command) error {
+	ngFilter := cmdutils.NewNodeGroupFilter()
+	ngFilter.SkipAll = withoutNodeGroup
+
+	if err := cmdutils.NewCreateClusterLoader(p, cfg, clusterConfigFile, nameArg, cmd, ngFilter).Load(); err != nil {
+		return err
+	}
+
+	if err := ngFilter.ValidateNodeGroupsAndSetDefaults(cfg.NodeGroups); err != nil {
+		return err
+	}
+
+	meta := cfg.Metadata
+	ctl := eks.New(p, cfg)
+
+	if !offline {
+		if err := ctl.CheckAuth(); err != nil {
+			return err
+		}
+	}
+
+	if !ctl.IsSupportedRegion() {
+		return cmdutils.ErrUnsupportedRegion(p)
+	}
+	logger.Info("using region %s", meta.Region)
+
+	if cfg.Metadata.Version == "" {
+		cfg.Metadata.Version = api.LatestVersion
+	}
+	if cfg.Metadata.Version != api.LatestVersion {
+		validVersion := false
+		for _, v := range api.SupportedVersions() {
+			if cfg.Metadata.Version == v {
+				validVersion = true
+			}
+		}
+		if !validVersion {
+			return fmt.Errorf("invalid version, supported values: %s", strings.Join(api.SupportedVersions(), ", "))
+		}
+	}
+	if cfg.Metadata.Region == "" {
+		cfg.Metadata.Region = api.DefaultRegion
+	}
+
+	// VPC and subnets
+	createOrImportVPC := func() error {
+
+		subnetInfo := func() string {
+			return fmt.Sprintf("VPC (%s) and subnets (private:%v public:%v)",
+				cfg.VPC.ID, cfg.PrivateSubnetIDs(), cfg.PublicSubnetIDs())
+		}
+
+		customNetworkingNotice := "custom VPC/subnets will be used; if resulting cluster doesn't function as expected, make sure to review the configuration of VPC/subnets"
+
+		canUseForPrivateNodeGroups := func(_ int, ng *api.NodeGroup) error {
+			if ng.PrivateNetworking && !cfg.HasSufficientPrivateSubnets() {
+				return fmt.Errorf("none or too few private subnets to use with --node-private-networking")
+			}
+			return nil
+		}
+
+		if !cfg.HasAnySubnets() {
+			// default: create dedicated VPC
+			if len(cfg.AvailabilityZones) == 0 {
+				cfg.AvailabilityZones = []string{
+					cfg.Metadata.Region + "a",
```

Comment:
yes, I've threw out a lot of complex code (not sure what kops is doing here anyway) because existing code was unreadable and replaced it with simple default to test it works for simple use case

please advise on the implementation that is acceptable and in scope for this PR

Reply:
for now I will make it static, just like region

Reply:
I agree, I also find it tricky to find the defaults for different parameters. We should take a look at that. If you have any ideas please let us know :)

Regarding the regions, let's see what @errordeveloper says.

Reply:
Why not use the AWS API to grab AZs for a region? https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-availability-zones.html

Reply:
Please use `pkg/az` for picking AZs, and we have `pkg/vpc` for picking subnets as well.